src/open_prime_rando/echoes/specific_area_patches/required_fixes.py

```diff
@@ -123,6 +129,37 @@ def main_research(editor: PatcherEditor):
     )
     _patch_echo_gate_softlock(area, 0x0B0315, relays)
 
+    area.get_layer("Contraption").active = False
+    portal_spawn = area.get_instance(0x0B0056)
+    spawn_xfm = portal_spawn.get_properties_as(SpawnPoint).editor_properties.transform
+
+    contraption_controller = area.get_layer("Default").add_instance_with(ScriptLayerController(
+        editor_properties=EditorProperties(
+            name="Dynamic Increment Contraption",
+            transform=spawn_xfm
+        ),
+        layer=LayerSwitch(
+            area_id=sanctuary_fortress.MAIN_RESEARCH_INTERNAL_ID,
+            layer_number=6
```

Comment:
Could use `.get_layer("...").index` so we have numbers :)

Reply:
oh yea true